src/rebase.rs

```diff
@@ -0,0 +1,339 @@
+    /// Returns none if the first operation has not yet been applied.
+    pub fn current_operation(&self) -> Option<RebaseOperation> {
+        if let Some(index) = self.current_operation_index() {
+            self.operation_at_index(index)
+        } else {
+            None
+        }
+    }
+}
+
+impl<'repo> Binding for Rebase<'repo> {
+    type Raw = *mut raw::git_rebase;
+    unsafe fn from_raw(raw: *mut raw::git_rebase) -> Rebase<'repo> {
+        Rebase {
+            raw: raw,
+            _marker: marker::PhantomData,
+        }
+    }
+    fn raw(&self) -> *mut raw::git_rebase {
+        self.raw
+    }
+}
+
+impl<'repo> Drop for Rebase<'repo> {
+    fn drop(&mut self) {
+        unsafe { raw::git_rebase_free(self.raw) }
+    }
+}
+
+impl<'rebase, 'repo: 'rebase> Binding for RebaseOperation<'rebase, 'repo> {
+    type Raw = *mut raw::git_rebase_operation;
+    unsafe fn from_raw(raw: *mut raw::git_rebase_operation) -> RebaseOperation<'rebase, 'repo> {
+        RebaseOperation {
+            raw: raw,
+            _marker: marker::PhantomData,
+        }
+    }
+    fn raw(&self) -> *mut raw::git_rebase_operation {
+        self.raw
+    }
+}
+
+impl<'a> RebaseOptions<'a> {
+    /// Initiate default set of rebase options.
+    pub fn new() -> RebaseOptions<'a> {
+        RebaseOptions {
+            version: raw::GIT_REBASE_OPTIONS_VERSION as usize,
+            quiet: false,
+            in_memory: false,
+            rewrite_notes_ref: None,
+            merge_options: None,
+            checkout_builder: None,
+        }
+    }
+
+    /// this will instruct other clients working
+    /// on this `rebase` that you want a quiet rebase experience.
+    /// This is provided for interoperability between Git tools
+    pub fn quiet(&mut self, quiet: bool) -> &mut Self {
+        self.quiet = quiet;
+        self
+    }
+
+    /// Perform an in-memory rebase, will not updated the repository to be in a rebasing-state
+    /// or modify the working directory.
+    pub fn in_memory(&mut self, in_memory: bool) -> &mut Self {
+        self.in_memory = in_memory;
+        self
+    }
+
+    /// The name of the notes reference used to rewrite notes
+    /// for rebased commits when finishing the rebase
+    pub fn rewrite_notes_ref(&mut self, rewrite_notes_ref: Option<CString>) -> &mut Self {
+        self.rewrite_notes_ref = rewrite_notes_ref;
+        self
+    }
+
+    /// Options to control how trees are merged during a `rebase`.
+    pub fn merge_options(&mut self, merge_options: Option<MergeOptions>) -> &mut Self {
+        self.merge_options = merge_options;
+        self
+    }
+
+    /// Options to control how files are written during a `rebase`.
+    pub fn checkout_builder(&mut self, checkout_builder: Option<CheckoutBuilder<'a>>) -> &mut Self {
+        self.checkout_builder = checkout_builder;
+        self
+    }
+
+    /// raw value of options
+    pub unsafe fn raw(&mut self) -> raw::git_rebase_options {
```

Comment:
I don't think it can, it needs to be exposed to the repo module. `MergeOptions` also exposes the raw method. `#[doc(hidden)]` could be added, then when `pub(crate)` becomes stable that could be used.